### ISSUE TITLE
Poor user experience on openshift credential logon if address space inadvertently created as system:admin

### ISSUE THREAD

**Issue by k-wall** [EnMasseProject/enmasse#1726]
If when creating my address space, I inadvertently run the `oc create -f ` as `system:admin` rather than say, developer, when I try to login to console with OpenShift credentials I see an terse error message that reads `{"error":"invalid_request","error_description":"Missing parameter: username"}`.

We should at very least document the error, and/or improve the error message.

**Comment by k-wall** [comment]
For completeness, when this occurs, the following is logged in the keycloak-controller log:

```
2018-10-03 12:27:05 WARN  ResourceChecker:53 - Exception in checker task
io.enmasse.user.model.v1.UserValidationFailedException: io.enmasse.user.model.v1.UserValidationFailedException: Invalid resource name 'standard.system:admin', must match ^[a-z]+([a-z0-9\-]*[a-z0-9]+|[a-z0-9]*)\.[a-z0-9]+([a-z0-9\-]*[a-z0-9]+|[a-z0-9]*)$
	at io.enmasse.user.model.v1.User.validate(User.java:61)
	at io.enmasse.user.keycloak.KeycloakUserApi.createUser(KeycloakUserApi.java:155)
	at io.enmasse.keycloak.controller.KeycloakManager.onUpdate(KeycloakManager.java:114)
	at io.enmasse.k8s.api.ResourceChecker.doWork(ResourceChecker.java:48)
	at io.enmasse.k8s.api.ResourceChecker.run(ResourceChecker.java:40)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.enmasse.user.model.v1.UserValidationFailedException: Invalid resource name 'standard.system:admin', must match ^[a-z]+([a-z0-9\-]*[a-z0-9]+|[a-z0-9]*)\.[a-z0-9]+([a-z0-9\-]*[a-z0-9]+|[a-z0-9]*)$
	at io.enmasse.user.model.v1.UserMetadata.validate(UserMetadata.java:70)
	at io.enmasse.user.model.v1.User.validate(User.java:58)
	... 5 more

```

It is possible to work around by creating a messaging user that federates to the openshift account.  It is necessary to get the uid of the openshift user first.

```
uid=$(oc get user developer  --output=go-template='{{.metadata.uid}}')
oc create -f - << EOF
apiVersion: user.enmasse.io/v1alpha1
kind: MessagingUser
metadata:
    name: standard.developer
spec:
    username: developer
    authentication:
      type: federated
      provider: openshift-v3
      federatedUserid: ${uid}
      federatedUsername: developer
    authorization:
    - addresses: [ "*" ]
      operations: [ "send", "recv", "view" ]
    - operations: [ "manage" ]
EOF

```

**Comment by k-wall** [comment]
Fixed by #2408.  OpenShift user is no longer required to exist as a federated user in the address space's keycloak instance.
